org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/AdapterUtils.java

```diff
@@ -0,0 +1,325 @@
+            if (new File(fullpath).isFile()) {
+                return fullpath;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Get java project from name.
+     * 
+     * @param projectName
+     *            project name
+     * @return java project
+     * @throws CoreException
+     *             CoreException
+     */
+    public static IJavaProject getJavaProjectFromName(String projectName) throws CoreException {
+        IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+        IProject project = root.getProject(projectName);
+        if (!project.exists()) {
+            throw new CoreException(new Status(IStatus.ERROR, JavaDebuggerServerPlugin.PLUGIN_ID, "Not an existed project."));
+        }
+        if (!project.isNatureEnabled("org.eclipse.jdt.core.javanature")) {
+            throw new CoreException(new Status(IStatus.ERROR, JavaDebuggerServerPlugin.PLUGIN_ID, "Not a project with java nature."));
+        }
+        IJavaProject javaProject = JavaCore.create(project);
+        return javaProject;
+    }
+
+    /**
+     * Get java project from type.
+     * 
+     * @param typeFullyQualifiedName
+     *            fully qualified name of type
+     * @return java project
+     * @throws CoreException
+     *             CoreException
+     */
+    public static List<IJavaProject> getJavaProjectFromType(String typeFullyQualifiedName) throws CoreException {
+        SearchPattern pattern = SearchPattern.createPattern(
+                typeFullyQualifiedName,
+                IJavaSearchConstants.TYPE,
+                IJavaSearchConstants.DECLARATIONS,
+                SearchPattern.R_EXACT_MATCH);
+        IJavaSearchScope scope = SearchEngine.createWorkspaceScope();
+        ArrayList<IJavaProject> projects = new ArrayList<>();
+        SearchRequestor requestor = new SearchRequestor() {
+            @Override
+            public void acceptSearchMatch(SearchMatch match) {
+                Object element = match.getElement();
+                if (element instanceof IJavaElement) {
+                    projects.add(((IJavaElement)element).getJavaProject());
+                }
+            }
+        };
+        SearchEngine searchEngine = new SearchEngine();
+        searchEngine.search(
+                pattern,
+                new SearchParticipant[] { SearchEngine.getDefaultSearchParticipant() },
+                scope,
+                requestor,
+                null /* progress monitor */);
+
+        return projects;
+    }
+
+    /**
+     * Gets java project from projectName and main class.
+     * @param projectName
+     *                   project name
+     * @param mainClass
+     *                   full qualified class name
+     * @return java project
+     * @throws CoreException
+     *                      CoreException
+     */
+    public static IJavaProject getJavaProject(String projectName, String mainClass) throws CoreException {
```

Comment:
i think this could be pending as later the interface wouldn't expose the method :)

---
In reply to: [130268214](https://github.com/vscjavaci/eclipse.jdt.ls/pull/23#discussion_r130268214) [](ancestors = 130268214)